runtime/src/sorted_storages.rs

```diff
@@ -227,17 +224,18 @@ pub mod tests {
         }
 
         pub fn new_for_tests(storages: &[&'a SnapshotStorage], slots: &[Slot]) -> Self {
-            SortedStorages::new_with_slots(storages.iter().cloned().zip(slots.iter()), None, None)
-        }
-
-        pub fn new_empty_for_tests() -> Self {
-            SortedStorages::new(&[])
+            assert_eq!(storages.len(), slots.len());
+            SortedStorages::new_with_slots(
+                storages.iter().cloned().zip(slots.iter().cloned()),
```

Comment:
I guess this clone might be what it takes to avoid a lifetime specifier, but is there any way to avoid it?

Reply:
this doesn't avoid the lifetime specifier. It is just there to get rid of references and make the types work. The clone is only in test code. Perhaps there is a rust way to get rid of the clone. I tried a few and the compiler and clippy both had opinions.